core/src/main/java/org/infinispan/distexec/DistributedExecutorService.java

```diff
@@ -25,12 +25,18 @@
  * return a result to task invoker.
  * <p>
  *
- *
  * Note that due to potential task migration to other nodes every {@link Callable},
  * {@link Runnable} and/or {@link DistributedCallable} submitted must be either {@link Serializable}
  * or {@link Externalizable}. Also the value returned from a callable must be {@link Serializable}
  * or {@link Externalizable}. Unfortunately if the value returned is not serializable then a
  * {@link NotSerializableException} will be thrown.
+ * <p>
+ *
+ * All {@link CompletableFuture} returned to the caller may be cancelled normally, however if interruption is desired
+ * to stop an in process task only the initial <b>CompletableFuture</b> may be used.  Any chained
```

Comment:
I wouldn't say anything about "normal" cancellation here, just that calling with `mayInterruptIfRunning == true` will interrupt the task on all the nodes where it is running, and link to `Thread#interrupt()`. Possibly also mention what  exception the user should expect when the task is cancelled, and that cancelling dependent `CompletableFutures` will not cancel this future (e.g. from `CompletableFuture.allOf()`).